### PR TITLE
Add new URI scheme descriptor for Promissory Notes

### PR DESCRIPTION
**Pull request by kkennis**
Add new `ReflectiveURIDescriptor.sol` contract that matches contract and token-based metadata URI scheme. Add tests. Perform deployment of new descriptor contracts for use on goerli and mainnet.

### CODE REVIEW

File: contracts/nft/ReflectiveURIDescriptor.sol [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+pragma solidity 0.8.18;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+import "../interfaces/INFTDescriptor.sol";
+
+/**
+ * @title BaseURIDescriptor
```
Comment (Mouzayan) [inline contracts/nft/ReflectiveURIDescriptor.sol:11]: ReflectiveURIDescriptor vs. BaseURIDescriptor.

File: contracts/nft/ReflectiveURIDescriptor.sol [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ * @author Non-Fungible Technologies, Inc.
+ *
+ * Basic descriptor contract for an NFT, that uses a baseURI, and returns a tokenURI
```
Comment (Mouzayan) [inline contracts/nft/ReflectiveURIDescriptor.sol:14]: Replace with description for ReflectiveURIDescriptor.

File: contracts/nft/ReflectiveURIDescriptor.sol [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    event SetBaseURI(address indexed caller, string indexed baseURI);
+
+    // ============================================ STATE ==============================================
+
+    string public baseURI;
+
+    // ========================================= CONSTRUCTOR ===========================================
+
+    /**
+     * @dev Creates a descriptor contract that allows a baseURI to be set,
```
Comment (Mouzayan) [inline contracts/nft/ReflectiveURIDescriptor.sol:29]: Do we need to add specifics re. ReflectiveURIDescriptor ?

File: contracts/nft/ReflectiveURIDescriptor.sol [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        // Empty baseURI is allowed
+        baseURI = _baseURI;
+    }
+
+    // ===================================== DESCRIPTOR OPERATIONAS ============================================
+
+    /**
+     * @notice Getter of specific URI for an ERC721 token ID.
+     *
+     * @param target              The address of the contract to get the URI for.
```
Comment (Mouzayan) [inline contracts/nft/ReflectiveURIDescriptor.sol:44]: `The` is not aligned with `The` on line 45.

File: scripts/deploy/set-note-descriptors.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    const { borrowerNote, lenderNote } = resources;
+
+    // Deploy a new descriptor
+    const descriptorFactory = await ethers.getContractFactory("ReflectiveURIDescriptor");
+    const descriptor = <ReflectiveURIDescriptor>await descriptorFactory.deploy(BASE_URI);
+    await descriptor.deployed();
+
+    console.log("ReflectiveURIDescriptor deployed to:", descriptor.address);
+
+    // Update notes
```
Comment (Mouzayan) [inline scripts/deploy/set-note-descriptors.ts:23]: Can delete 23 - 28.

File: scripts/deploy/set-note-descriptors.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    // let tx = await borrowerNote.connect(admin).setDescriptor(descriptor.address);
+    // await tx.wait();
+
+    // tx = await lenderNote.connect(admin).setDescriptor(descriptor.address);
+    // await tx.wait();
+
+    console.log("✅ Upgraded descriptor contracts.");
```
Comment (Mouzayan) [inline scripts/deploy/set-note-descriptors.ts:30]: `Deployed ReflectiveURIDescriptor` vs. `Upgraded descriptor`?

**Review by Mouzayan — COMMENTED** [review]
Looks good. Few nits found below.

File: contracts/nft/ReflectiveURIDescriptor.sol [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+     *      within token URIs enumerated from the base.
+     *
+     * @param _baseURI              The value of the baseURI state variable.
+     */
+    constructor(string memory _baseURI) {
+        // Empty baseURI is allowed
+        baseURI = _baseURI;
+    }
+
+    // ===================================== DESCRIPTOR OPERATIONAS ============================================
```
Comment (PowVT) [inline contracts/nft/ReflectiveURIDescriptor.sol:39]: Typo, looks like it is in the base and static as well :/

**Review by PowVT — APPROVED** [review]
Nice, just seeing natspec updates. I verified the contract on Goerli as well.

**Pull request merged**